harness/determined/common/experimental/experiment.py

```diff
@@ -115,6 +119,49 @@ def reload(self) -> None:
         resp = bindings.get_GetExperiment(self._session, experimentId=self.id).experiment
         self._hydrate(resp)
 
+    def add_label(self, label: str) -> None:
+        """Add a label to the experiment.
+
+        Arguments:
+            label: a string label to add to the experiment. If the label already exists,
+                the method call will be a no-op.
+        """
+
+        # URL-encode label for request.
+        label = parse.quote(label)
+        resp = bindings.put_PutExperimentLabel(
+            session=self._session, experimentId=self.id, label=label
+        )
+        self.labels = list(resp.labels)
+
+    def remove_label(self, label: str) -> None:
+        """Removes a label from the experiment.
+
+        Arguments:
+            label: a string label to remove from the experiment. If the specified label does not
+                exist on the experiment, this method call will be a no-op.
+        """
+        # URL-encode label for request.
+        label = parse.quote(label)
+        resp = bindings.delete_DeleteExperimentLabel(
+            session=self._session, experimentId=self.id, label=label
+        )
+        self.labels = list(resp.labels)
+
+    def set_labels(self, labels: List[str]) -> None:
+        """Sets experiment labels to the specified list.
+
+        This method will overwrite any existing labels on the experiment with the specified
+        labels.
+
+        Arguments:
+            labels: a list of string labels to set on the experiment. Duplicates will be ignored.
+        """
+        patch_exp = bindings.v1PatchExperiment(id=self.id, labels=labels)
+        resp = bindings.patch_PatchExperiment(self._session, body=patch_exp, experiment_id=self.id)
+        assert resp.experiment
```

Comment:
we need `Unset`, i think probably because we have no good way to tell the difference between `None` and "undefined" values, as in whether it was explicitly set to None, or not set at all.